session7/src/com/itschool/homework/HomeRunChallenge13.java

```diff
@@ -0,0 +1,110 @@
+package com.itschool.homework;
+
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Scanner;
+
+public class HomeRunChallenge13 {
+    public static void main(String[] args) {
+        // apelez metoda generateMaze si atribui rezultatul ei intr-o variabila de tip matrice
+        String[][] maze = generateMaze();
+        // verific rezultatul returnat de apelul functiei BFS
+        if (BFS(maze)) {
+            System.out.println("There is a a path from S to E");
+        } else {
+            System.out.println("No solutiion found");
+        }
+    }
+
+    // metoda ce returneaza o matrice de siruri de caractere si o afiseaza
+    public static String[][] generateMaze() {
+        Scanner input = new Scanner(System.in);
+        String[][] maze = new String[4][4];
+        System.out.println("Please give the elements of the maze consisting by 0 and 1");
```

Comment:
it is pretty hard to give a valid input here because your input accepts each matrix element on different lines.. I understand you want this input to be dynamic and inserted by user, but it hard for the user to predict a valid input here. You can hard-code the maze instead, so at line 22 something like:
`String[][] maze = { {"1", "0", "0", "1"}, {"0", "0", "1", "1"}, {"1", "0", "0", "1"}, {"0", "1", "0", "0"} };`
and then change values here for other tests.